feat: [M3-6427] - Add No Results section for Marketplace Search

cpathipa opened the pull request:
## Description 📝
Add No Results section for Marketplace Search

## Preview 📷
![image](https://user-images.githubusercontent.com/119517080/232818144-ee181b17-3a4b-4f67-8aa4-9adec4cbaa5a.png)


**Remove this section or include a screenshot or screen recording of the change**

## How to test 🧪

- Navigate to Marketplace search
- Search for app that doesn't exists (ex; datadog)
- Should display "No Results" message. 

**How do I run relevant unit or e2e tests?**

- yarn test
- yarn cy:run

jdamore-linode commented:
Nice, it works well!

Would it be possible to center the "No Results" text?

<img width="1351" alt="Screen Shot 2023-04-13 at 4 11 29 PM" src="https://user-images.githubusercontent.com/97627410/231872385-ba542ccf-3995-48f6-8b2f-807fd221e74d.png">

Do we need any other input from the UX team as far as wording and design should go?

mjac0bs reviewed (commented):
I'm glad we're doing this! Seconding @jdamore-linode's request to center the text horizontally. And I'm wondering about the overall placement within that container. 

Mock (assume actual center alignment):
![Screenshot 2023-04-13 at 3 09 00 PM](https://user-images.githubusercontent.com/114685994/231894758-cd147aef-f33e-445f-878f-f8a831343ed7.jpg)

cpathipa commented:
Thank you @jdamore-linode and @mjac0bs thank you for the feedback, I will circle back with UX on this.

dwiley-akamai reviewed (commented):
I also agree with checking on the centered-within-container option

cpathipa commented:
Thank you for the feedback. Based on the UX feedback, updated the no results copy and adjusted alignment to center. 
![image](https://user-images.githubusercontent.com/119517080/232560166-728e0e87-7239-40e7-b9a2-cb45d48a9a64.png)

hana-linode reviewed (commented):
Thoughts on left aligning the no results section? The selection cards are left aligned as well as the "no compatible images" text below

![image](https://user-images.githubusercontent.com/115299789/232583293-ebc50f94-d4d6-4003-b5d9-4e2487917945.png)

mjac0bs commented:
Nitpicks on copy: I'm wondering if clearer copy might read: "Sorry, no results matched your search for "datadog". Please try another term."

Thought process being:
1. "no <search term> results" sounds kind of strange and is slightly harder to scan as a reader than "no results"
2. telling the user to "please try again" feels misleading because if the user were to try the same action again, the results would be the same. It feels more clear to state what they should actually try to yield different results.

What do others (and UX) think?

hana-linode commented:
@mjac0bs Agreed that "Sorry, no results matched your search for "datadog" sounds clearer. I don't think we even need the second sentence as it's pretty obvious that the user should search for a different term

jaalah-akamai commented on packages/manager/src/features/linodes/LinodesCreate/AppPanelSection.tsx:
```diff
@@ -46,38 +48,46 @@ export const AppPanelSection: React.FC<Props> = (props) => {
+              {`Sorry, no results matching "${searchValue}" were found.`}
+            </Typography>
+          </Grid>
+        )}
```
I think we can simplify the JSX a little here by doing:
```
{apps.length > 0 ? (
  <AppPanelGrid container spacing={2}>
    {apps.map((eachApp) => { ... })}
  </AppPanelGrid>
) : (
  <Typography>
    {`Sorry, no results matching "${searchValue}" were found.`}
  </Typography>
)}
```
We won't need the extra Grid in this case

jaalah-akamai commented:
Let's remember to update changelog too 👍

mjac0bs reviewed (approved):
Just a changelog update and this looks like it's good to go.
